frontend/src/components/board/HorizontalMenu.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC } from "react";
+import { MenuGroup } from "../../interfaces/canva-interfaces";
+import { MenuItem } from "@/interfaces/canva-interfaces";
+import { CogIcon } from "lucide-react";
+import Image from "next/image";
+import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
+import {
+  HoverCard,
+  HoverCardContent,
+  HoverCardTrigger,
+} from "@/components/ui/hover-card";
+
+interface HorizontalMenuProps {
+  menuItem: MenuGroup;
+  onIconClick: (name: string) => void;
+  withSettings?: boolean;
+  fromRight?: boolean;
+  boardName: string;
+  onActiveItemChange?: (activeItem: string | null) => void;
+  activeItem: string | null;
+  activeCanvasObject: { [key: string]: any } | null;
+}
+
+const HorizontalMenu: FC<HorizontalMenuProps> = ({
+  menuItem,
+  onIconClick,
+  withSettings = false,
+  fromRight = false,
+  boardName,
+  onActiveItemChange,
+  activeItem,
+  activeCanvasObject,
+}) => {
+  const handleClick = (name: string, action?: () => void) => {
+    if (onActiveItemChange) {
+      onActiveItemChange(name);
+    }
+    if (action) {
+      action();
+    }
+    onIconClick(name);
+  };
+
+  return (
+    <div
+      className={`flex items-center justify-between  px-4 bg-white border-b pl-0`}
+    >
+      <div className="flex">
+        <div className="border-r flex items-center justify-center h-[64px] h-min-[64px] w-[3.5em]">
+          <Image src="/best-logo-lite.png" alt="Logo" width={20} height={20} />
+        </div>
+        <div className="flex-shrink-0  flex items-center border-r px-6">
+          <span className="text-lg font-semibold">{boardName}</span>
+        </div>
+        <div className="flex items-center overflow-x-auto space-x-2 px-4">
+          {menuItem.items.map((item: MenuItem, itemIndex: number) => {
+            if(!(activeCanvasObject && activeCanvasObject.type === 'activeSelection') && (item.name === 'group-selected')) return null;
+            else if(!(activeCanvasObject) && (item.name === 'remove-selected')) return null;
+            return (<HoverCard key={itemIndex}>
```

Comment:
Let's add a new line here :)
```suggestion
            return (
            <HoverCard key={itemIndex}>
```